lib/api/apiUtils/object/getReplicationInfo.js

```diff
@@ -63,14 +64,21 @@ function _getReplicationInfo(rule, replicationConfig, content, operationType,
  * @param {boolean} objSize - The size, in bytes, of the object being PUT
  * @param {string} operationType - The type of operation to replicate
  * @param {object} objectMD - The object metadata
+ * @param {AuthInfo} [authInfo] - authentication info of object owner
  * @return {undefined}
  */
 function getReplicationInfo(objKey, bucketMD, isMD, objSize, operationType,
-    objectMD) {
+    objectMD, authInfo) {
     const content = isMD || objSize === 0 ? ['METADATA'] : ['DATA', 'METADATA'];
     const config = bucketMD.getReplicationConfiguration();
     // If bucket does not have a replication configuration, do not replicate.
     if (config) {
+        // If delete an object due to a lifecycle action,
```

Comment:
Shouldn't we check for operation type? In theory nothing lifecycle does should be replicated (unless configured, seems AWS has that option now), but we're targeting deletes from expiration right now - while an object that we PUT from a transition workflow would be expected to replicate if a policy is present.

Reply:
I think Lifecycle Transition is using backbeatclient.multipleBackendPutObject() request to “PUT object”. 
The request then gets proxied to CloudServer and  Cloudserver uses its own “multiple backend” clients to put the object in the right transitioned location. 
So Cloudserver won’t be able to retrieve the "lifecycle assumed role" session name in that case and the PUT should pass the check and succeed.
Also, I am not sure if it is necessary to add a safeguarding check that checks for “DELETE” method or checks that `isDeleteMarker` property is true.

Reply:
https://github.com/scality/backbeat/blob/development/8.3/extensions/replication/tasks/CopyLocationTask.js#L289

Reply:
I don't know if the backbeat-specific handlers on cloudserver would understand any types of credentials besides old service account (assuming they would), but backbeat's transition is only using old-style credentials because we haven't made the change yet, in the next few weeks in artesca we'll do the same for transition as we did for expiration to use assumed role credentials. So in theory they would match lifecycle actions.

For DELETE op vs delete marker, I'm not sure, it seems plain DELETEs should never be replicated, and delete markers should be replicated only if not from lifecycle (described [here](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-what-is-isnot-replicated.html#replication-what-is-replicated) as we don't use the new-format replication configuration) - so checking for `isDeleteMarker` in conjunction with lifecycle would be enough.